products/BellHybrid/apps/application-bell-main/presenters/HomeScreenPresenter.cpp

```diff
@@ -24,23 +24,7 @@ namespace app::home_screen
                                              std::unique_ptr<AbstractTimeModel> timeModel)
         : app{app}, alarmModel{std::move(alarmModel)}, batteryModel{std::move(batteryModel)},
           temperatureModel{std::move(temperatureModel)}, timeModel{std::move(timeModel)}
-    {
-        constexpr int timeout = pdMS_TO_TICKS(1500);
-
-        auto response =
-            app->bus.sendUnicastSync(std::make_shared<sevm::LatchStatusRequest>(), service::name::evt_manager, timeout);
-
-        if (response.first == sys::ReturnCodes::Success) {
-            auto msgState = dynamic_cast<sevm::LatchStatusResponse *>(response.second.get());
-            if (msgState == nullptr) {
-                return;
-            }
-
-            if (msgState->getStatus() == sevm::LatchStatus::PRESSED) {
-                latchPressed = true;
-            }
-        }
-    }
```

Comment:
Please look at this ticket: https://appnroll.atlassian.net/browse/BH-1180
Won't removing this latch check in the HomeScreenPresenter constructor affect that functionality required in BH-1180?

Reply:
It shouldn't. The home screen state machine was modified to properly calculate the initial state based on both the current state of the latch and alarm model info. Each time, a user enters the home screen the state machine reloads and its initial state is recalculated.

Here's the change:
```
                                           "Init"_s + event<Events::ModelReady> [Helpers::shouldSwitchToSnooze] = "AlarmSnoozed"_s,
                                             "Init"_s + event<Events::ModelReady> [Helpers::shouldSwitchToRinging] = "AlarmRinging"_s,
                                             "Init"_s + event<Events::ModelReady> [Helpers::shouldSwitchToDeactivated] = "Deactivated"_s,
                                             "Init"_s + event<Events::ModelReady> [Helpers::shouldSwitchToActivated] = "Activated"_s,
```